esphome/components/tsl2591/sensor.py

```diff
@@ -0,0 +1,155 @@
+    "9876X": TSL2591Gain.TSL2591_GAIN_MULTIPLIER_MAX,
+    "MAX": TSL2591Gain.TSL2591_GAIN_MULTIPLIER_MAX,
+    "MAXIMUM": TSL2591Gain.TSL2591_GAIN_MULTIPLIER_MAX,
+}
+
+
+def validate_integration_time(value):
+    value = cv.positive_time_period_milliseconds(value).total_milliseconds
+    return cv.enum(INTEGRATION_TIMES, int=True)(value)
+
+
+TSL2591Component = tsl2591_ns.class_(
+    "TSL2591Component", cg.PollingComponent, i2c.I2CDevice
+)
+
+
+CONFIG_SCHEMA = (
+    cv.Schema(
+        {
+            cv.GenerateID(): cv.declare_id(TSL2591Component),
+            cv.Optional(CONF_INFRARED): sensor.sensor_schema(
+                UNIT_LUX,
+                ICON_BRIGHTNESS_6,
+                0,
+                DEVICE_CLASS_ILLUMINANCE,
+                STATE_CLASS_MEASUREMENT,
+            ),
+            cv.Optional(CONF_VISIBLE): sensor.sensor_schema(
+                UNIT_LUX,
+                ICON_BRIGHTNESS_6,
+                0,
+                DEVICE_CLASS_ILLUMINANCE,
+                STATE_CLASS_MEASUREMENT,
+            ),
+            cv.Optional(CONF_FULL_SPECTRUM): sensor.sensor_schema(
+                UNIT_LUX,
+                ICON_BRIGHTNESS_6,
+                0,
+                DEVICE_CLASS_ILLUMINANCE,
+                STATE_CLASS_MEASUREMENT,
+            ),
+            cv.Optional(CONF_CALCULATED_LUX): sensor.sensor_schema(
+                UNIT_LUX,
+                ICON_BRIGHTNESS_6,
+                4,
+                DEVICE_CLASS_ILLUMINANCE,
+                STATE_CLASS_MEASUREMENT,
+            ),
+            cv.Optional(
+                CONF_INTEGRATION_TIME, default="100ms"
+            ): validate_integration_time,
+            cv.Optional(CONF_GAIN, default="MEDIUM"): cv.enum(GAINS, upper=True),
+        }
+    )
+    .extend(cv.polling_component_schema("60s"))
+    .extend(i2c.i2c_device_schema(0x29))
+)
+
+
+async def to_code(config):
+    var = cg.new_Pvariable(config[CONF_ID])
+    await cg.register_component(var, config)
+    await i2c.register_i2c_device(var, config)
+
+    if CONF_FULL_SPECTRUM in config:
+        conf = config[CONF_FULL_SPECTRUM]
+        sens = await sensor.new_sensor(conf)
+        cg.add(var.set_full_spectrum_sensor(sens))
+
+    if CONF_INFRARED in config:
+        conf = config[CONF_INFRARED]
+        sens = await sensor.new_sensor(conf)
+        cg.add(var.set_infrared_sensor(sens))
+
+    if CONF_VISIBLE in config:
+        conf = config[CONF_VISIBLE]
+        sens = await sensor.new_sensor(conf)
+        cg.add(var.set_visible_sensor(sens))
+
+    if CONF_CALCULATED_LUX in config:
+        conf = config[CONF_CALCULATED_LUX]
+        sens = await sensor.new_sensor(conf)
+        cg.add(var.set_calculated_lux_sensor(sens))
+
+    cg.add(var.set_integration_time(config[CONF_INTEGRATION_TIME]))
+    cg.add(var.set_gain(config[CONF_GAIN]))
+    # https://platformio.org/lib/show/463/Adafruit%20TSL2591%20Library
+    cg.add_library("Adafruit TSL2591 Library", "^1.4.0")
```

Comment:
```suggestion
    cg.add_library("adafruit/Adafruit TSL2591 Library", "^1.4.0")
```

PlatformIO wants libraries to have an owner nowadays :)